src/note_encryption.rs

```diff
@@ -209,29 +208,18 @@ impl Domain for OrchardDomain {
         plaintext: &[u8],
     ) -> Option<(Self::Note, Self::Recipient)> {
         orchard_parse_note_plaintext_without_memo(self, plaintext, |diversifier| {
-            Some(DiversifiedTransmissionKey::derive(ivk, diversifier))
+            DiversifiedTransmissionKey::derive(ivk, diversifier)
         })
     }
 
     fn parse_note_plaintext_without_memo_ovk(
         &self,
         pk_d: &Self::DiversifiedTransmissionKey,
-        esk: &Self::EphemeralSecretKey,
-        ephemeral_key: &EphemeralKeyBytes,
+        _esk: &Self::EphemeralSecretKey,
+        _ephemeral_key: &EphemeralKeyBytes,
```

Comment:
```suggestion
```
See https://github.com/zcash/librustzcash/pull/848/files#r1199198303

It may seem as though this is making life more difficult because removing these arguments is a breaking change, but strictly speaking the change to the guarantee of the function is a breaking API change anyway, even if not one that would otherwise be detected by the Rust compiler.

That is, it would be incorrect to mix a version of `orchard` from after this PR with a version of `zcash_note_encryption` from before https://github.com/zcash/librustzcash/pull/848 &mdash; not (as it happens) because it would be insecure per se, but because that usage hasn't been reviewed.

Reply:
Mixing in the way you describe cannot happen, because the traits would not be version-compatible. Rust code is never dynamically linked internally.